frontend/hooks/useBuilds.js

```diff
@@ -0,0 +1,60 @@
+import { useQuery, useQueryClient, useMutation } from '@tanstack/react-query';
+import api from '@util/federalistApi';
+
+// Assumes the order of builds is based on createdAt desc
+export function getLatestBuildByBranch(builds) {
+  const latestBranches = [];
+
+  return builds.map((build) => {
+    if (build.completedAt && !latestBranches.includes(build.branch)) {
+      latestBranches.push(build.branch);
+      return { ...build, latestForBranch: true };
+    }
+
+    return { ...build, latestForBranch: false };
+  });
+}
+
+export function useBuilds(siteId) {
+  const { isPending, error, data } = useQuery({
+    queryKey: ['builds', parseInt(siteId, 10)],
+    queryFn: () =>
+      api.fetchBuilds({ id: siteId }).then((builds) => getLatestBuildByBranch(builds)),
+    refetchInterval: 10000,
+    refetchIntervalInBackground: false,
+  });
+
+  return { isPending, error, data };
+}
+
+export function useRebuild(siteId, buildId, ref) {
+  const queryClient = useQueryClient();
+
+  const mutation = useMutation({
+    mutationFn: ({ buildId, siteId }) => api.restartBuild(buildId, siteId),
+    onSuccess: () => {
+      // Scroll to build history container ref
+      // Invalidate and refetch
+      return Promise.all([
+        ref.current.scrollIntoView({
```

Comment:
this scroll is nice when the build is far down but does a confusing jolt if it's in the first 1-3. Is there an easy way to make it conditional?

Reply:
I added an offset limit between the `window.scrollY` and element top.